File: Median_of_Two_Sorted_Arrays.py
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        x=sorted(nums1+nums2)
        n=len(x)
        if n%2==0:
            y=float(x[n//2-1]+x[n//2])
            return float(y/2)
        else:
            return float(x[n//2])

nums1=[1,2]
nums2=[3]

sol=Solution()
print(sol.findMedianSortedArrays(nums1,nums2))

File: Median_of_Two_Sorted_Arrays.java
import java.util.Arrays;

class Solution
{
    double findMedianSortedArrays(int[] nums1, int[] nums2)
    {
        int[] x=new int[nums1.length+nums2.length];
        System.arraycopy(nums1, 0, x, 0, nums1.length);
        System.arraycopy(nums2,0, x, nums1.length, nums2.length);
        Arrays.sort(x);
        int n=x.length;
        if (n%2==0)
        {
            return (x[n/2-1]+x[n/2])/2.0;
        }
        else
        {
            return (x[n/2]);
        }
    }
}

public class Median_of_Two_Sorted_Arrays {
    public static void main(String[] args)
    {
        int[] nums1={1,3};
        int[] nums2={2,4};
        Solution m=new Solution();
        System.out.println(m.findMedianSortedArrays(nums1, nums2));
    }
}

